### PR TITLE
feat(explorer): change party to pubkey

### LINKED ISSUE #2866: Use public key instead of/alongside party id in explorer
The block explorer references 'party id', but users may expect to see the phrase 'public key'. Introduce the 'public key' terminology instead of (and perhaps alongside, in some places) in order to make that connection clearer.

### PR DESCRIPTION
**Pull request by edd**
# Related issues 🔗

Closes #2866

# Description ℹ️

In most places, 'public key' is a better label than 'party id'

### CODE REVIEW

File: apps/explorer/src/app/routes/parties/id/components/party-id-error.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ type PartyIdErrorProps = {
 const PartyIdError = ({ id, error }: PartyIdErrorProps) => {
   const end = isValidPartyId(id)
     ? t('No accounts or transactions found for: ')
-    : 'Invalid party id: ';
+    : 'Invalid pubkey: ';
```
Comment (sam-keen) [inline apps/explorer/src/app/routes/parties/id/components/party-id-error.tsx:17]: Suggestion (non-blocking): for consistency's sake, this one should be 'public key' too

**Review by sam-keen — APPROVED** [review]

File: apps/explorer/src/app/routes/parties/id/components/party-id-error.tsx (hunk 1 above)
Comment (candida-d) [reply apps/explorer/src/app/routes/parties/id/components/party-id-error.tsx:17]: Agreed!

**Review by candida-d — APPROVED** [review]
But I second Sam's comment about pubkey vs public key

**Review by mattrussell36 — APPROVED** [review]

File: apps/explorer/src/app/routes/parties/id/components/party-id-error.tsx (hunk 1 above)
Comment (edd) [inline apps/explorer/src/app/routes/parties/id/components/party-id-error.tsx:17]:
```suggestion
    : 'Invalid public key: ';
```

**Pull request merged**